### ISSUE TITLE
Modal does not close on confirm?

### ISSUE THREAD

**Issue by jochendaum** [agoragames/confirm-with-reveal#13]
Hi,

I've downloaded confirm_with_reveal.js just now and implemented like this:

- Foundation from here: https://packagist.org/packages/bmatzner/foundation-bundle
- <script src="/bundles/bmatznerfoundation/js/foundation/foundation.js"></script>
    <script src="/bundles/bmatznerfoundation/js/foundation/foundation.reveal.js"></script>
    <script src="/bundles/agrihealthahp/js/offlineHandler.js"></script>   <-- thats mine
    <script src="/bundles/agrihealthahp/js/confirm_with_reveal.js"></script>

- <div class="row form_actions delete hidden" style="display: block;">
                <div class="small-12 medium-offset-7 medium-2 columns end">
                    <button data-confirm="{"ok":"Yup","cancel":"Nope"}">Delete</button>
                </div>
            </div>

I catch the confirm event with:

jQuery('#' + model + '_edit .form_actions.delete').show().on('confirm.reveal', null, { name: model, rowId: rowId }, call.deleteRecord );

which works fine. However the modal does not close, I'm using 

jQuery('.reveal-modal').hide();
jQuery('.reveal-modal-bg').hide();

to close it manually.

It is closing on Cancel.

Any idea why  this would be happening?

**Comment by jletourneau** [comment]
Currently, the plugin does not close the dialog when the confirmation button is pressed — this was by design, for what I felt was the more common case of these dialogs triggering full page refreshes as opposed to Ajax or front-end MVC updates (the idea being that leaving the dialog up while the new page loads prevents any other actions from being taken in the mean time if the response is not immediate).

I will see about adding a configuration option for this behavior, probably something along the lines of `close_after_ok` or the like, for cases where people want to use this to confirm actions that don't entail a page refresh.

**Comment by jochendaum** [comment]
Cheers, I can live with that. However it it was to close by default, the people using it to link to other pages  would not have any downside either?

**Comment by jletourneau** [comment]
The only downside to closing the dialog by default when the OK button is hit is that if the dialog is confirming (for example) deletion of a resource, or some other important action, and the server response takes more than a few tenths of a second to refresh the page in the user's browser, the user may mistakenly think that the action was carried out via Ajax, and navigate off the page before the response arrives.

I recognize the value of having a `close_after_ok` option and will look into adding that, but I think the safer or more conservative setting is to have that behavior as a configuration option that must be explicitly chosen by the developer using the plugin.

**Comment by MandicaAtSupsi** [comment]
Fixed with this in code on 49 line, before:
if ($el.is('form, :input')) {
    return $el.closest('form').removeAttr('data-confirm').submit();
}

After:
if ($el.is('form, :input')) {
    **modal.foundation('reveal', 'close');**
    return $el.closest('form').removeAttr('data-confirm').submit();
}

**Comment by jekuno** [comment]
I agree closing the modal when clicking confirm should become the deafult. (Nowadays most apps are SPA i.e. there's no full page reload after clicking confirm.)
However IMHO `modal.foundation('reveal', 'close');` should be one line above (outside of the if clause).
